runtime/Go/antlr/lexer.go

```diff
@@ -25,11 +27,12 @@ type Lexer interface {
 type BaseLexer struct {
 	*BaseRecognizer
 
-	Interpreter         *LexerATNSimulator
+	Interpreter         ILexerATNSimulator
 	TokenStartCharIndex int
 	TokenStartLine      int
 	TokenStartColumn    int
 	ActionType          int
+	Virt                Lexer // The most derived lexer implementation. Allows virtual method calls.
```

Comment:
The idiomatic way is to embed the "super class" instance in the "subclass" instance and define the same method in the subclass. Will that not work here?